### PR TITLE
Database.load() checks which file to load

### PR DESCRIPTION
**Pull request by hagenw**
Checks if it should load from PKL or CSV file.
If PKL file exists it is preferred unless a newer CSV file exists.

### CODE REVIEW

**Comment by hagenw** [comment]
#36 should be merged first, before we continue here.

**Comment by hagenw** [comment]
I updated and fixed the tests.

File: audformat/core/table.py [hunk 1]
```diff
@@ -540,6 +540,11 @@ def load(
     ):
         r"""Load table data from disk.
 
+        Tables can be stored as PKL and/or CSV files to disk.
+        It will load from the PKL file
+        if available and its modification date is newer,
+        otherwise the CSV file is loaded.
```
Comment (frankenjoe) [inline audformat/core/table.py:546]: Should we raise a warning if the modification date of the CSV file is newer than that of the PKL file?

Comment (frankenjoe) [reply audformat/core/table.py:546]: Most likely this means the two are not in sync, right?

Comment (hagenw) [reply audformat/core/table.py:546]: yes, but I would not raise a warning then, because it might be totally on purpose, e.g. you modified the CSV file.

Comment (frankenjoe) [reply audformat/core/table.py:546]: yes, but if I do that I should either update the PKL or delete it

Comment (frankenjoe) [reply audformat/core/table.py:546]: or you think with this approach we are safe enough that it can never fire back if CSV and PKL are not in sync?

Comment (hagenw) [reply audformat/core/table.py:546]: This was the whole idea of this approach, to make it more safe.

Another solution would be to raise an error an ask the user to first delete the PKL file.

Comment (frankenjoe) [reply audformat/core/table.py:546]: Yes, that would be the safest solution in my opinion.

Comment (hagenw) [reply audformat/core/table.py]: I changed it. It now raises an error if the CSV file is newer.

File: audformat/core/table.py [hunk 2]
```diff
@@ -556,7 +564,27 @@ def load(
...
+                    and os.path.getmtime(csv_file) > os.path.getmtime(pkl_file)
+            ):
+                print('CSV time:', os.path.getmtime(csv_file))
+                print('PKL time:', os.path.getmtime(pkl_file))
+                raise RuntimeError(
+                    f"The table CSV file '{csv_file}' is newer "
+                    f"than the table PKL file '{pkl_file}'. "
+                    "If you want to load from the CSV file, "
+                    "please delete the PKL file. "
+                    "If you wanto to load from the PKL file, "
```
Comment (frankenjoe) [inline audformat/core/table.py:583]:
```suggestion
                    "If you want to load from the PKL file, "
```

**Pull request merged**